lib/base64.c

```diff
@@ -196,51 +195,41 @@ static CURLcode base64_encode(const char *table64,
   if(!output)
     return CURLE_OUT_OF_MEMORY;
 
-  while(insize > 0) {
+  while(insize >= 3) {
+    *output++ = table64[ in[0] >> 2 ];
+    *output++ = table64[ ((in[0] & 0x03) << 4) | (in[1] >> 4) ];
+    *output++ = table64[ ((in[1] & 0x0F) << 2) | ((in[2] & 0xC0) >> 6) ];
+    *output++ = table64[ in[2] & 0x3F ];
+    insize -= 3;
+    in += 3;
+  }
+  if(insize) {
+    /* this is only one or two bytes now */
```

Comment:
at this point you could unroll it and get rid of inputparts and i

```suggestion
    /* this is only one or two bytes now */
    unsigned char ibuf[2];
    ibuf[0] = *in++;
    insize--;
    if(insize) {
      ibuf[1] = *in++;
     --insize;
    }
    else
      ibuf[1] = 0;
```

then later
~~~c
    if(!ibuf[1]) {
      if(*padstr) {
        ...
~~~

Reply:
yes, I did that in a subsequent commit!